container/container.go

```diff
@@ -1761,3 +1894,79 @@ func (b *ContainerWorker) ContainersMatchingAgreement(agreements []string, inclu
+	// find the ms from the local db, it is okay if the ms def is not found. this is old behavious befor the ms split.
+	if msdef, err := persistence.FindMicroserviceDef(b.db, api_spec, version); err != nil {
+		return nil, fmt.Errorf("Error finding microservice definition from the local db for %v version %v. %v", api_spec, version, err)
+	} else if msdef != nil && msdef.Workloads != nil && len(msdef.Workloads) > 0 {
+		// get the service name from the ms def
+		for _, wl := range msdef.Workloads {
+			deploymentDesc := new(DeploymentDescription)
+			if err := json.Unmarshal([]byte(wl.Deployment), &deploymentDesc); err != nil {
+				return nil, fmt.Errorf("Error Unmarshalling deployment string %v for microservice %vversion %v. %v", wl.Deployment, api_spec, version, err)
+			} else {
+				for serviceName, _ := range deploymentDesc.Services {
+					container_names = append(container_names, serviceName)
+				}
+			}
+		}
+	}
+	glog.V(5).Infof("The container names for microservice %v version %v are: %v", api_spec, version, container_names)
+	return container_names, nil
+}
+
+// go through all the microservice containers for the given microservices and check if the containers are up and running.
+// If yes, return the list of containers
+func (b *ContainerWorker) findMicroserviceContainers(agreementId string, microservices []events.MicroserviceSpec) ([]docker.APIContainers, error) {
+	ms_containers := make([]docker.APIContainers, 0)
+	if containers, err := b.client.ListContainers(docker.ListContainersOptions{}); err != nil {
+		return nil, fmt.Errorf("Unable to get list of running containers: %v", err)
+	} else {
+		for _, api_spec := range microservices {
+			// find the ms from the local db,
+			if msc_names, err := b.findMicroserviceDefContainerNames(api_spec.SpecRef, api_spec.Version); err != nil {
+				return nil, fmt.Errorf("Error finding microservice definition from the local db for %v. %v", api_spec, err)
+			} else if msc_names == nil || len(msc_names) == 0 {
+				glog.V(5).Infof("No containers defined for microdevice %v", api_spec)
+				continue
+			} else if msinsts, err := persistence.FindMicroserviceInstances(b.db, []persistence.MIFilter{persistence.AllInstancesMIFilter(api_spec.SpecRef, api_spec.Version), persistence.UnarchivedMIFilter()}); err != nil {
+				return nil, fmt.Errorf("Error retrieving microservice instances for %v version %v from database, error: %v", api_spec.SpecRef, api_spec.Version, err)
+			} else if msinsts == nil || len(msinsts) == 0 {
+				return nil, fmt.Errorf("Microservice instance has not be initiated for microservice  %v yet.", api_spec)
+			} else {
+				// use the first instance for now, assuming there is a single instance for ms.  Will change after the ms muti instance support.
+				ms_instance := msinsts[0]
+
+				// save the agreement id to the microservice instance
+				if persistence.UpdateMSInstanceAssociaedAgreements(b.db, ms_instance.GetKey(), true, agreementId); err != nil {
```

Comment:
Why are you saving the agreement id to the instance at this point? Isn't the purpose of this function just to verify that all the needed instances are up so that we can start the workload?